src/Controller/CRUDController.php

```diff
@@ -1360,8 +1360,27 @@ protected function handleXmlHttpRequestErrorResponse(Request $request, FormInter
         }
 
         $errors = [];
-        foreach ($form->getErrors(true) as $error) {
-            $errors[] = $error->getMessage();
+        $formName = $form->getName();
+
+        foreach ($form->getErrors(true) as $formError) {
+            $origin = $formError->getOrigin();
+
+            $fieldName = $origin->getName();
+            $name = '';
+
+            while ($origin = $origin->getParent()) {
```

Comment:
It's the same logic than https://github.com/symfony/symfony/blob/6.2/src/Symfony/Component/Form/Extension/DataCollector/FormDataExtractor.php#L146

I would recommend to
- use the id key of `extractConfiguration` https://github.com/symfony/symfony/blob/6.2/src/Symfony/Component/Form/Extension/DataCollector/FormDataExtractor.php#L28
- or maybe try to use the `extractSubmittedData` which has an errors key https://github.com/symfony/symfony/blob/6.2/src/Symfony/Component/Form/Extension/DataCollector/FormDataExtractor.php#L87

Reply:
We can't directly use those DataCollector, those are used for collect data to show in the Profiler, not sure if that is a good idea.

I will start looking at this again, any idea is appreciated. I was thinking on looking how api platform does it.

Reply:
Why cant we use it ?

This is collecting errors and formatting them ; this seemed a good idea to use it...

Reply:
They are not loaded on the container unless you enable debug mode.